client/http_sender.go

```diff
@@ -99,90 +92,60 @@ func (s *httpSender) Send(_ context.Context, call proto.Call) {
 	retryOpts := s.retryOpts
 	retryOpts.Tag = fmt.Sprintf("%s %s", s.context.RequestScheme(), call.Method())
 
-	if err := retry.WithBackoff(retryOpts, func() (retry.Status, error) {
-		resp, err := s.post(call)
-		if err != nil {
-			if resp != nil {
-				infoErr := util.Errorf("failed to send HTTP request with %s", err)
-				// See if we can retry based on HTTP response code.
-				switch resp.StatusCode {
-				case http.StatusServiceUnavailable, http.StatusGatewayTimeout, StatusTooManyRequests:
-					// Retry on service unavailable and request timeout.
-					// TODO(spencer): consider respecting the Retry-After header for
-					// backoff / retry duration.
-					return retry.Continue, infoErr
-				default:
-					// Can't recover from all other errors.
-					return retry.Break, infoErr
-				}
-			}
-			switch err.(type) {
-			case *httpSendError:
-				// Assume all errors sending request are retryable. The actual
-				// number of things that could go wrong is vast, but we don't
-				// want to miss any which should in theory be retried with
-				// the same client command ID. We log the error here as a
-				// warning so there's visiblity that this is happening. Some of
-				// the errors we'll sweep up in this net shouldn't be retried,
-				// but we can't really know for sure which.
-				log.Warningf("failed to send HTTP request or read its response: %s", err)
-				return retry.Continue, nil
-			default:
-				// Can't retry in order to recover from this error. Propagate.
-				return retry.Break, err
-			}
-		}
-		// On successful post, we're done with retry loop.
-		return retry.Break, nil
-	}); err != nil {
-		call.Reply.Header().SetGoError(err)
-	}
-}
-
-// post posts the call using the HTTP client. The call's method is
-// appended to KVDBEndpoint and set as the URL path. The call's arguments
-// are protobuf-serialized and written as the POST body. The content
-// type is set to application/x-protobuf.
-//
-// On success, the response body is unmarshalled into call.Reply.
-func (s *httpSender) post(call proto.Call) (*http.Response, error) {
 	// Marshal the args into a request body.
 	body, err := gogoproto.Marshal(call.Args)
 	if err != nil {
-		return nil, err
+		return err
```

Comment:
`Send` doesn't return a value. How is this compiling? Or am I misreading the code?